File: readability.c
#include <stdio.h>
#include <cs50.h>
#include <math.h>

int main (void)
{
    string text;
    do
    {
        text = get_string("Text: ");
    }
    while (text[0] == '\0');
    
    float cont_l = 0;
    int letter;
    float cont_w = 1;
    float cont_s = 0;
    for (letter = 0; text[letter] != '\0'; letter ++) //reviews everyletter
    {
        if ((text[letter] >= 'a' && text[letter] <= 'z') | (text[letter] >= 'A' && text[letter] <= 'Z'))
        {
            cont_l ++; //letter counter
        } 
        
        if (text[letter] == ' ')
        {
            cont_w ++; //word counter
        }
        
        if (text[letter] == '.' | text[letter] == '!' | text[letter] == '?')
        {
            cont_s ++; //sentence counter
        }
    }

    float L = ((cont_l / cont_w) * 100); // average number of letters per 100 words in the text
    
    float S = ((cont_s / cont_w) * 100); //average number of sentences per 100 words in the text.
    
    float grade = (0.0588 * L - 0.296 * S - 15.8); //Coleman-Liau formula
    
    if (grade >= 16)     // printer of grades
    {
        printf("Grade 16+\n");
    }
    else if (grade < 16 && grade >= 1)
    {
        printf("Grade %.0f \n", round(grade));
    }
    else if (grade < 1)
    {
        printf("Before Grade 1\n");
    }
    
}

File: readability.py
from cs50 import get_string

while True:
    
    text = get_string("Text: ")
    
    if text != "" or text != " ":
        break

cont_l = 0.0  # letter counter
letter = 0  # letters
cont_w = 1.0  # word counter
cont_s = 0.0  # sentences counter 
L = 0.0 
S = 0.0
grade = 0

for letter in text:  # letter by letter
    
    if (letter >= 'a' and letter <= 'z') or (letter >= 'A' and letter <= 'Z'):
        cont_l += 1  # couhts letter
        
    if letter == ' ':
        cont_w += 1  # counts words
        
    if letter == '.' or letter == '!' or letter == '?':
        cont_s += 1  # counts sentences
        
L = ((cont_l / cont_w) * 100)  # average number of letters per 100 words in the text

S = ((cont_s / cont_w) * 100)  # average number of sentences per 100 words in the text.

grade = round(0.0588 * L - 0.296 * S - 15.8)  # Coleman-Liau formula

if grade >= 16:
    print("Grade 16+")  # final grade
elif grade < 16 and grade >= 1:
    print("Grade", grade)  # final grade
elif grade < 1:
    print("Before grade 1")  # final grade
    

